### PR TITLE
[FEAT] Deposit

### PR DESCRIPTION
**Pull request by spencerdezartsmith**
# Description
Co-Authored by: [Romy Furtado](<25412194+Rofrtd@users.noreply.github.com>)

**NOTE: You will need to add the next deposit contract address to your `.env` to test**

This PR adds a new in-game deposit system. You can access this from the same locations you could access before. You're able to see the balances of SFL and Collectibles in your personal and add what you would like to send to your farm.

You will no longer be required to manually "refresh" to see the deposited items on your farm. 

Screen Shot 2023-02-16 at 9 23 00 am

Screen Shot 2023-02-16 at 9 41 58 am

Fixes #issue

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

- Click on Balance on your farm and deposit SFL and Items.
- Go to Goblin Retreat and click on the Bank and do another deposit.

NOTE: You should not be able to access the deposit modal from the Community Garden

# Checklist:

- [x] Title of the PR is relevant and is prefixed with [FEAT], [CHORE] or [FIX]
- [x] Screenshot if it includes any UI changes
- [x] I have read the contributing guidelines and agree to the T&Cs
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by adamhannigan — DISMISSED** [review]
Works well ✅ Code looks good to me 🏆 

Just a few suggestions:

1. Now that we explicitly support the deposit flow it may be useful to include a link "Deposit didn't arrive?" that we can link to our docs. There we can explain the "Refresh" button in the settings if needed.
2. In Goblin Retreat the Balance widget looks like it can be clicked (cursor pointer) but nothing happens. I feel we should support the same mechanism as the main island and open the deposit modal when clicked on
3. Can we include a CTA to deposit from the inventory UI as well? It may be useful to clear confusion when people buy items on a secondary market and don't see them in their basket or chest. A small link "Deposit from personal wallet" or something similar that pops open the modal.

File: src/features/goblins/bank/components/Deposit.tsx [hunk 1]
```diff
@@ -1,48 +1,266 @@
...
+                    </div>
+                  )}
+                </div>
+              </div>
+            </div>
+            <div className="mb-1 mt-2">
+              <a
+                target="_blank"
+                className="underline text-xxs hover:text-blue-500"
+                href={`https://app.gitbook.com/o/bntEYvEP4dzoxNUbvnHA/s/IWGEodCG0c07OUY9iW9t/~/changes/297/economy/depositing-and-custody#cant-see-the-items-you-deposited`}
```
Comment (adamhannigan) [inline src/features/goblins/bank/components/Deposit.tsx:248]: Just a reminder to update this to the docs.sunflower-land.com link before merging ;)

**Review by adamhannigan — APPROVED** [review]
Flow feels great! Well done ⭐

File: src/features/goblins/bank/components/Deposit.tsx (hunk 1 above)
Comment (spencerdezartsmith) [reply src/features/goblins/bank/components/Deposit.tsx:248]: Already done mate.

**Pull request merged**